### ISSUE TITLE
新版本Xinference 部署glm4-chat，launch webui 报错

### ISSUE THREAD

**Issue by lergliu** [xorbitsai/inference#2041]
### System Info / 系統信息

accelerate                    0.33.0
aiobotocore                   2.7.0
aiofiles                      23.2.1
aiohappyeyeballs              2.3.5
aiohttp                       3.10.1
aioitertools                  0.11.0
aioprometheus                 23.12.0
aiosignal                     1.3.1
altair                        5.3.0
annotated-types               0.7.0
anyio                         4.4.0
async-timeout                 4.0.3
attrdict                      2.0.1
attrs                         24.2.0
auto_gptq                     0.7.1
autoawq                       0.2.6
autoawq_kernels               0.0.7
bcrypt                        4.2.0
bitsandbytes                  0.43.3
botocore                      1.31.64
certifi                       2024.7.4
cffi                          1.17.0
charset-normalizer            3.3.2
click                         8.1.7
cloudpickle                   3.0.0
colorama                      0.4.6
coloredlogs                   15.0.1
contourpy                     1.2.1
cryptography                  43.0.0
cycler                        0.12.1
datasets                      2.20.0
dill                          0.3.8
distro                        1.9.0
ecdsa                         0.19.0
einops                        0.8.0
fastapi                       0.110.3
ffmpy                         0.4.0
filelock                      3.15.4
fonttools                     4.53.1
frozenlist                    1.4.1
fsspec                        2023.10.0
gekko                         1.2.1
gradio                        4.26.0
gradio_client                 0.15.1
h11                           0.14.0
httpcore                      1.0.5
httpx                         0.27.0
huggingface-hub               0.24.5
humanfriendly                 10.0
idna                          3.7
importlib_resources           6.4.0
Jinja2                        3.1.4
jiter                         0.5.0
jmespath                      1.0.1
jsonschema                    4.23.0
jsonschema-specifications     2023.12.1
kiwisolver                    1.4.5
markdown-it-py                3.0.0
MarkupSafe                    2.1.5
matplotlib                    3.9.1.post1
mdurl                         0.1.2
modelscope                    1.17.0
mpmath                        1.3.0
multidict                     6.0.5
multiprocess                  0.70.16
networkx                      3.3
numpy                         1.26.4
nvidia-cublas-cu12            12.1.3.1
nvidia-cuda-cupti-cu12        12.1.105
nvidia-cuda-nvrtc-cu12        12.1.105
nvidia-cuda-runtime-cu12      12.1.105
nvidia-cudnn-cu12             8.9.2.26
nvidia-cufft-cu12             11.0.2.54
nvidia-curand-cu12            10.3.2.106
nvidia-cusolver-cu12          11.4.5.107
nvidia-cusparse-cu12          12.1.0.106
nvidia-nccl-cu12              2.20.5
nvidia-nvjitlink-cu12         12.6.20
nvidia-nvtx-cu12              12.1.105
openai                        1.39.0
opencv-contrib-python         4.10.0.84
optimum                       1.21.3
orjson                        3.10.6
packaging                     24.1
pandas                        2.2.2
passlib                       1.7.4
peft                          0.12.0
pillow                        10.4.0
pip                           24.0
protobuf                      5.27.3
psutil                        6.0.0
pyarrow                       17.0.0
pyarrow-hotfix                0.6
pyasn1                        0.6.0
pycparser                     2.22
pydantic                      2.8.2
pydantic_core                 2.20.1
pydub                         0.25.1
Pygments                      2.18.0
pynvml                        11.5.3
pyparsing                     3.1.2
python-dateutil               2.9.0.post0
python-jose                   3.3.0
python-multipart              0.0.9
pytz                          2024.1
PyYAML                        6.0.2
quantile-python               1.1
referencing                   0.35.1
regex                         2024.7.24
requests                      2.32.3
rich                          13.7.1
rouge                         1.0.1
rpds-py                       0.20.0
rsa                           4.9
ruff                          0.5.6
s3fs                          2023.10.0
safetensors                   0.4.4
scipy                         1.14.0
semantic-version              2.10.0
sentencepiece                 0.2.0
setuptools                    72.1.0
shellingham                   1.5.4
six                           1.16.0
sniffio                       1.3.1
sse-starlette                 2.1.3
starlette                     0.37.2
sympy                         1.13.1
tabulate                      0.9.0
tblib                         3.0.0
tiktoken                      0.7.0
timm                          1.0.8
tokenizers                    0.19.1
tomlkit                       0.12.0
toolz                         0.12.1
torch                         2.3.1
torchvision                   0.18.1
tqdm                          4.66.5
transformers                  4.43.4
transformers-stream-generator 0.0.5
triton                        2.3.1
typer                         0.11.1
typing_extensions             4.12.2
tzdata                        2024.1
urllib3                       2.0.7
uvicorn                       0.30.5
uvloop                        0.19.0
websockets                    11.0.3
wheel                         0.43.0
wrapt                         1.16.0
xinference                    0.14.0.post1
xoscar                        0.3.2
xxhash                        3.4.1
yarl                          1.9.4
zstandard                     0.23.0

### Running Xinference with Docker? / 是否使用 Docker 运行 Xinfernece？

- [ ] docker / docker
- [X] pip install / 通过 pip install 安装
- [ ] installation from source / 从源码安装

### Version info / 版本信息

0.14.0.post1

### The command used to start Xinference / 用以启动 xinference 的命令

xinference-local --host 0.0.0.0 --port 9997

### Reproduction / 复现过程

2024-08-08 10:25:20,544 xinference.api.restful_api 32805 ERROR    Handling request http://127.0.0.1:9997/v1/ui/custom-glm4-chat failed: 1 validation error for BuildGradioInterfaceRequest
model_description
  none is not an allowed value (type=type_error.none.not_allowed)
Traceback (most recent call last):
  File "/root/anaconda3/envs/xinference2/lib/python3.11/site-packages/starlette/middleware/errors.py", line 164, in __call__
    await self.app(scope, receive, _send)
  File "/root/anaconda3/envs/xinference2/lib/python3.11/site-packages/aioprometheus/asgi/middleware.py", line 184, in __call__
    await self.asgi_callable(scope, receive, wrapped_send)
  File "/root/anaconda3/envs/xinference2/lib/python3.11/site-packages/starlette/middleware/cors.py", line 93, in __call__
    await self.simple_response(scope, receive, send, request_headers=headers)
  File "/root/anaconda3/envs/xinference2/lib/python3.11/site-packages/starlette/middleware/cors.py", line 148, in simple_response
    await self.app(scope, receive, send)
  File "/root/anaconda3/envs/xinference2/lib/python3.11/site-packages/starlette/middleware/exceptions.py", line 65, in __call__
    await wrap_app_handling_exceptions(self.app, conn)(scope, receive, send)
  File "/root/anaconda3/envs/xinference2/lib/python3.11/site-packages/starlette/_exception_handler.py", line 64, in wrapped_app
    raise exc
  File "/root/anaconda3/envs/xinference2/lib/python3.11/site-packages/starlette/_exception_handler.py", line 53, in wrapped_app
    await app(scope, receive, sender)
  File "/root/anaconda3/envs/xinference2/lib/python3.11/site-packages/starlette/routing.py", line 756, in __call__
    await self.middleware_stack(scope, receive, send)
  File "/root/anaconda3/envs/xinference2/lib/python3.11/site-packages/starlette/routing.py", line 776, in app
    await route.handle(scope, receive, send)
  File "/root/anaconda3/envs/xinference2/lib/python3.11/site-packages/starlette/routing.py", line 297, in handle
    await self.app(scope, receive, send)
  File "/root/anaconda3/envs/xinference2/lib/python3.11/site-packages/starlette/routing.py", line 77, in app
    await wrap_app_handling_exceptions(app, request)(scope, receive, send)
  File "/root/anaconda3/envs/xinference2/lib/python3.11/site-packages/starlette/_exception_handler.py", line 64, in wrapped_app
    raise exc
  File "/root/anaconda3/envs/xinference2/lib/python3.11/site-packages/starlette/_exception_handler.py", line 53, in wrapped_app
    await app(scope, receive, sender)
  File "/root/anaconda3/envs/xinference2/lib/python3.11/site-packages/starlette/routing.py", line 72, in app
    response = await func(request)
               ^^^^^^^^^^^^^^^^^^^
  File "/root/anaconda3/envs/xinference2/lib/python3.11/site-packages/fastapi/routing.py", line 278, in app
    raw_response = await run_endpoint_function(
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/root/anaconda3/envs/xinference2/lib/python3.11/site-packages/fastapi/routing.py", line 191, in run_endpoint_function
    return await dependant.call(**values)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/root/anaconda3/envs/xinference2/lib/python3.11/site-packages/xinference/api/restful_api.py", line 951, in build_gradio_interface
    body = BuildGradioInterfaceRequest.parse_obj(payload)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/root/anaconda3/envs/xinference2/lib/python3.11/site-packages/pydantic/v1/main.py", line 526, in parse_obj
    return cls(**obj)
           ^^^^^^^^^^
  File "/root/anaconda3/envs/xinference2/lib/python3.11/site-packages/pydantic/v1/main.py", line 341, in __init__
    raise validation_error
pydantic.v1.error_wrappers.ValidationError: 1 validation error for BuildGradioInterfaceRequest
model_description
  none is not an allowed value (type=type_error.none.not_allowed)

### Expected behavior / 期待表现

正常运行

**Comment by Mysterious0o0** [comment]
Is this problem solved? I also encountered this problem

**Comment by Bookstall** [comment]
## TL;DR

Check your model configuration json file to see if some corresponding fields (e.g. `model_description`) are missing.

The required fields is defined in `BuildGradioInterfaceRequest`:

```python
# https://github.com/xorbitsai/inference/blob/main/xinference/api/restful_api.py#L187
class BuildGradioInterfaceRequest(BaseModel):
    model_type: str
    model_name: str
    model_size_in_billions: int
    model_format: str
    quantization: str
    context_length: int
    model_ability: List[str]
    model_description: str
    model_lang: List[str]
```

## System and version info

I used the Ubuntu 20.04 with python 3.10, and some packages' info like:

```shell
pydantic                          2.8.2
pydantic_core                     2.20.1
pydantic-settings                 2.5.2
torch                             2.3.1
transformers                      4.44.2
xinference                        0.15.2
```

## Analysis

When i launch the chat webui, i got the error like:

```shell
2024-09-20 18:38:54,922 xinference.api.restful_api 11698 ERROR    Handling request http://127.0.0.1:9997/v1/ui/custom-glm-4-9b-chat failed: 1 validation error for BuildGradioInterfaceRequest
model_description
  none is not an allowed value (type=type_error.none.not_allowed)
Traceback (most recent call last):
  File "/opt/conda/envs/xinference/lib/python3.10/site-packages/starlette/middleware/errors.py", line 164, in __call__
    await self.app(scope, receive, _send)
  File "/opt/conda/envs/xinference/lib/python3.10/site-packages/aioprometheus/asgi/middleware.py", line 184, in __call__
    await self.asgi_callable(scope, receive, wrapped_send)
  File "/opt/conda/envs/xinference/lib/python3.10/site-packages/starlette/middleware/cors.py", line 93, in __call__
    await self.simple_response(scope, receive, send, request_headers=headers)
  File "/opt/conda/envs/xinference/lib/python3.10/site-packages/starlette/middleware/cors.py", line 148, in simple_response
    await self.app(scope, receive, send)
  File "/opt/conda/envs/xinference/lib/python3.10/site-packages/starlette/middleware/exceptions.py", line 65, in __call__
    await wrap_app_handling_exceptions(self.app, conn)(scope, receive, send)
  File "/opt/conda/envs/xinference/lib/python3.10/site-packages/starlette/_exception_handler.py", line 64, in wrapped_app
    raise exc
  File "/opt/conda/envs/xinference/lib/python3.10/site-packages/starlette/_exception_handler.py", line 53, in wrapped_app
    await app(scope, receive, sender)
  File "/opt/conda/envs/xinference/lib/python3.10/site-packages/starlette/routing.py", line 756, in __call__
    await self.middleware_stack(scope, receive, send)
  File "/opt/conda/envs/xinference/lib/python3.10/site-packages/starlette/routing.py", line 776, in app
    await route.handle(scope, receive, send)
  File "/opt/conda/envs/xinference/lib/python3.10/site-packages/starlette/routing.py", line 297, in handle
    await self.app(scope, receive, send)
  File "/opt/conda/envs/xinference/lib/python3.10/site-packages/starlette/routing.py", line 77, in app
    await wrap_app_handling_exceptions(app, request)(scope, receive, send)
  File "/opt/conda/envs/xinference/lib/python3.10/site-packages/starlette/_exception_handler.py", line 64, in wrapped_app
    raise exc
  File "/opt/conda/envs/xinference/lib/python3.10/site-packages/starlette/_exception_handler.py", line 53, in wrapped_app
    await app(scope, receive, sender)
  File "/opt/conda/envs/xinference/lib/python3.10/site-packages/starlette/routing.py", line 72, in app
    response = await func(request)
  File "/opt/conda/envs/xinference/lib/python3.10/site-packages/fastapi/routing.py", line 278, in app
    raw_response = await run_endpoint_function(
  File "/opt/conda/envs/xinference/lib/python3.10/site-packages/fastapi/routing.py", line 191, in run_endpoint_function
    return await dependant.call(**values)
  File "/opt/conda/envs/xinference/lib/python3.10/site-packages/xinference/api/restful_api.py", line 1061, in build_gradio_interface
    body = BuildGradioInterfaceRequest.parse_obj(payload)
  File "/opt/conda/envs/xinference/lib/python3.10/site-packages/pydantic/v1/main.py", line 526, in parse_obj
    return cls(**obj)
  File "/opt/conda/envs/xinference/lib/python3.10/site-packages/pydantic/v1/main.py", line 341, in __init__
    raise validation_error
pydantic.v1.error_wrappers.ValidationError: 1 validation error for BuildGradioInterfaceRequest
model_description
  none is not an allowed value (type=type_error.none.not_allowed)
```

According the traceback infomation, we can check the key code in `xinference/api/restful_api.py`. Then we added a `print` sentence to get the `payload` information:

```python
# https://github.com/xorbitsai/inference/blob/main/xinference/api/restful_api.py#L187
class BuildGradioInterfaceRequest(BaseModel):
    model_type: str
    model_name: str
    model_size_in_billions: int
    model_format: str
    quantization: str
    context_length: int
    model_ability: List[str]
    model_description: str # required field（and must not None）
    model_lang: List[str]

# https://github.com/xorbitsai/inference/blob/main/xinference/api/restful_api.py#L1052
async def build_gradio_interface(
	self, model_uid: str, request: Request
) -> JSONResponse:
	"""
	Separate build_interface with launch_model
	build_interface requires RESTful Client for API calls
	but calling API in async function does not return
	"""
	payload = await request.json()
	
        # the print sentence we added
	print(f"build_gradio_interface's payload: {payload}") 

	body = BuildGradioInterfaceRequest.parse_obj(payload)
	assert self._app is not None
	assert body.model_type == "LLM"
```

Then we relaunch the xinference service by `xinference-local`, we can see the `payload` infomation:

```shell
build_gradio_interface's payload: {'model_type': 'LLM', 'model_name': 'glm4-chat', 'model_size_in_billions': 9, 'model_format': 'pytorch', 'quantization': 'none', 'context_length': 131072, 'model_ability': ['chat', 'tools'], 'model_description': None, 'model_lang': ['en', 'zh']}

2024-09-20 19:26:31,028 xinference.api.restful_api 23484 ERROR    Handling request http://127.0.0.1:9997/v1/ui/custom-glm-4-9b-chat failed: 1 validation error for BuildGradioInterfaceRequest
model_description
  none is not an allowed value (type=type_error.none.not_allowed)
```

**And we found the `model_description` field is `None`, which makes it impossible to pass Pydantic's data verification. It is the reason why we cannot launch the chat webui.** 

![Xinference-model-description-is-None](https://github.com/user-attachments/assets/c4a57ff2-73b2-4fed-8a53-c7b5447a18ec)

## Solve method

The solve method is simple, we only to add the related filed (in here is `model_description`) in the `custom-glm-4-9b-chat.json` file:

```json
{
        "model_ability": [
	    "chat",
	    "tools"
	  ],
	
	"model_description": "GLM4 is the open source version of the latest generation of pre-trained models in the GLM-4 series launched by Zhipu AI.",
	
        "model_family": "glm4-chat",
}
```

And then we terminate, unregister, register, launch the model:

```shell
$ xinference terminate --model-uid custom-glm-4-9b-chat

$ xinference unregister --model-type LLM --model-name custom-glm-4-9b-chat

$ xinference register --model-type LLM --file ./model-json/custom-glm-4-9b-chat.json --persist --endpoint http://127.0.0.1:9997

$ xinference launch --model-engine Transformers --model-name custom-glm-4-9b-chat --model-format pytorch
```

![Xinference-model-description-is-not-None](https://github.com/user-attachments/assets/ad8a2ea2-8d69-4b11-93f8-7423eb25e66a)

Now we can successfully relaunch the chat webui, and auto jump to the http://127.0.0.1:9997/custom-glm-4-9b-chat/

![Xinference-launch-chat-ui](https://github.com/user-attachments/assets/c40b2097-f5c2-4740-ab74-19adb9d1e051)

## More

- [官方文档](https://inference.readthedocs.io/zh-cn/latest/models/custom.html#define-a-custom-llm-model) 在自定义 `custom-llama-2-chat` 时，并没有添加 `model_description` 字段

- 从 `v0.14.0` 版本开始，如果你需要注册的模型的家族是 **Xinference 内置支持的模型**，你可以直接通过 `launch` 接口中的 `model_path` 参数来启动它，从而免去注册步骤的麻烦。现在非常推荐使用这种方式。具体信息参见 [官方文档](https://inference.readthedocs.io/zh-cn/latest/models/custom.html#directly-launch-an-existing-model)
